Iphone12LAMP_TEST/Listings/config.txt
; generated by Component: ARM Compiler 5.06 update 6 (build 750) Tool: ArmCC [4d3637]
; commandline ArmCC [--c99 --list --split_sections --debug -c --asm --interleave -o.\objects\config.o --asm_dir=.\Listings\ --list_dir=.\Listings\ --depend=.\objects\config.d --cpu=Cortex-M0 --apcs=interwork -O1 --diag_suppress=9931 -I.\APP -I.\APP\TN2115s_libs -I.\Common\CMSIS-FreeRTOS\10.0.1\Source\include -I.\Common\RTE\RTOS -I.\Common\CMSIS-FreeRTOS\10.0.1\CMSIS\RTOS2\FreeRTOS\Include -I.\Common\CMSIS-FreeRTOS\10.0.1\Source\portable\RVDS\ARM_CM0 -I.\APP\EINK -I.\Device_desc -I.\Common\CMSIS-FreeRTOS\10.0.1\Source -I.\Common\RTE\Device\ARMCM0 -I.\RTE\_soc_m0 -IC:\Users\bbb\AppData\Local\Arm\Packs\ARM\CMSIS\5.6.0\CMSIS\Core\Include -IC:\Users\bbb\AppData\Local\Arm\Packs\ARM\CMSIS\5.6.0\Device\ARM\ARMCM0\Include -D__MICROLIB -D__UVISION_VERSION=529 -D_RTE_ -DARMCM0 --omf_browse=.\objects\config.crf APP\TN2115s_libs\config.c]
                          THUMB

                          AREA ||.ARM.__AT_0x00001FFC||, DATA, READONLY, ALIGN=2

                  rdp_word0
                          DCD      0x00000000

                          AREA ||.ARM.__AT_0x00002000||, DATA, ALIGN=0

                  FW_VER_0
000000  4f637420          DCB      0x4f,0x63,0x74,0x20
000004  32392032          DCB      0x32,0x39,0x20,0x32
000008  30323031          DCB      0x30,0x32,0x30,0x31
00000c  303a3238          DCB      0x30,0x3a,0x32,0x38
000010  3a333400          DCB      0x3a,0x33,0x34,0x00

                          AREA ||.ARM.__AT_0x00002020||, DATA, ALIGN=2

                  app_addr
                          DCD      __main

                          AREA ||.ARM.__AT_0x00003FFC||, DATA, READONLY, ALIGN=2

                  rdp_word1
                          DCD      0x00000000

                          AREA ||.ARM.__AT_0x00005FFC||, DATA, READONLY, ALIGN=2

                  rdp_word2
                          DCD      0x00000000

                          AREA ||.ARM.__AT_0x00007FD0||, DATA, ALIGN=0

                  FW_VER_1
000000  4f637420          DCB      0x4f,0x63,0x74,0x20
000004  32392032          DCB      0x32,0x39,0x20,0x32
000008  30323031          DCB      0x30,0x32,0x30,0x31
00000c  303a3238          DCB      0x30,0x3a,0x32,0x38
000010  3a333400          DCB      0x3a,0x33,0x34,0x00

                          AREA ||.ARM.__AT_0x00007FFC||, DATA, READONLY, ALIGN=2

                  rdp_word3
                          DCD      0x00000000

                          AREA ||.ARM.__AT_0x60000780||, DATA, READONLY, ALIGN=0

                  def_tag
000000  4653544e          DCB      0x46,0x53,0x54,0x4e
000004  31306d20          DCB      0x31,0x30,0x6d,0x20
000008  64480000          DCB      0x64,0x48,0x00,0x00
00000c  e110fc00          DCB      0xe1,0x10,0xfc,0x00
000010  0303d000          DCB      0x03,0x03,0xd0,0x00
000014  00fe0000          DCB      0x00,0xfe,0x00,0x00
000018  00000000          DCB      0x00,0x00,0x00,0x00
00001c  00000000          DCB      0x00,0x00,0x00,0x00

                          AREA ||.data||, DATA, ALIGN=2

                  rc_setting_frq
                          DCD      0x600007a0

                          AREA ||area_number.12||, DATA, ALIGN=2

                          EXPORTAS ||area_number.12||, ||.data||
                  def_sreg
                          DCD      0x600007f0

                          AREA ||area_number.13||, DATA, ALIGN=2

                          EXPORTAS ||area_number.13||, ||.data||
                  SystemCoreClock
                          DCD      0x007a1200

                          AREA ER_IROM1, DATA, ALIGN=2

                  tag_in_ram
                          DCD      0x00000000

;*** Start embedded assembler ***

#line 1 "APP\\TN2115s_libs\\config.c"
	AREA ||.rev16_text||, CODE
	THUMB
	EXPORT |__asm___8_config_c_def_tag____REV16|
#line 492 "C:\\Users\\bbb\\AppData\\Local\\Arm\\Packs\\ARM\\CMSIS\\5.6.0\\CMSIS\\Core\\Include\\cmsis_armcc.h"
|__asm___8_config_c_def_tag____REV16| PROC
#line 493

 rev16 r0, r0
 bx lr
	ENDP
	AREA ||.revsh_text||, CODE
	THUMB
	EXPORT |__asm___8_config_c_def_tag____REVSH|
#line 507
|__asm___8_config_c_def_tag____REVSH| PROC
#line 508

 revsh r0, r0
 bx lr
	ENDP

;*** End   embedded assembler ***


Iphone12LAMP_TEST/APP/TN2115s_libs/config.c
#include "project.h"

/**********************************Configurations***************************************/

#ifdef RDP_ACTIVE
#undef RDP_ACTIVE
#endif

#ifdef TAG_MEM_CFG
#undef TAG_MEM_CFG
#endif

/*System ROM read-out protection switch. Uncomment to turn on.*/
//#define RDP_ACTIVE																																		

/*TAG eeprom default values, should follow Type2 Tag memory layout rules. */
const uint8_t def_tag[32]  __attribute__((at(0x60000780))) = {											
												0x46, 0x53, 0x54, 0x4e, 0x31, 0x30, 0x6d, 0x20,
												0x64, 0x48, 0x00, 0x00, 0xe1, 0x10, 0xfc, 0x00,
												0x03, 0x03, 0xd0, 0x00, 0x00, 0xfe, 0x00, 0x00,
												0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
};

/*Configure TAG memory location, 0 = EEPROM, 1 = RAM */
#define		TAG_MEM_CFG			0

/*FW version is set here, maximum 20bytes long. The two version strings must match*/
/*Each updated version must have an unique FW version value. This can be read out in OTA mode.*/
volatile uint8_t FW_VER_0[] __attribute__((at(0x00002000))) = __DATE__ __TIME__;
volatile uint8_t FW_VER_1[] __attribute__((at(0x00007FD0))) = __DATE__ __TIME__;



/**********************************DO NOT MODIFY BELOW***************************************/

#ifdef	RDP_ACTIVE
#define RDP_WORD		0xffffffff
#else
#define RDP_WORD		0x00000000
#endif

#if (TAG_MEM_CFG == 0)
uint8_t	 *	tag_in_ram 	__attribute__((section("ER_IROM1")))	= 0;
#elif (TAG_MEM_CFG == 1)
uint32_t		tag_mem[16] __attribute__( ( section( "NoInit"),zero_init) );	
uint8_t	 *	tag_in_ram 	__attribute__((section("ER_IROM1"))) 	= (uint8_t *)tag_mem;
#endif

const uint32_t rdp_word0 __attribute__((at(0x00001FFC))) = RDP_WORD;
const uint32_t rdp_word1 __attribute__((at(0x00003FFC))) = RDP_WORD;
const uint32_t rdp_word2 __attribute__((at(0x00005FFC))) = RDP_WORD;
const uint32_t rdp_word3 __attribute__((at(0x00007FFC))) = RDP_WORD;

extern void __main(void);
const void * app_addr __attribute__((at(0x000002020))) = (void *)__main;

uint16_t * rc_setting_frq = (uint16_t *)TAG_RCCAL_BASE;
uint8_t * def_sreg = (uint8_t *)TAG_REG_EEPROM_BASE;

uint32_t SystemCoreClock = 8000000UL;
